crates/pallet-domains/src/staking.rs

```diff
@@ -711,10 +711,62 @@ pub(crate) fn do_deregister_operator<T: Config>(
     })
 }
 
+/// Different type of withdrawal
+///
+/// NOTE: if the deposit was made in the current epoch, the user may not be able to withdraw it
+// until the current epoch ends
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub enum WithdrawStake<Balance, Share> {
+    /// Withdraw all stake
+    All,
+    /// Withdraw a given percentage of the stake
+    Percent(Percent),
+    /// Withdraw a given amount of stake, calculated by the share price at
+    /// this instant, it may not be accurate and may withdraw a bit more
+    /// stake if a reward happens later in this epoch
+    Stake(Balance),
+    /// Withdraw a given amount of share
+    Share(Share),
+}
+
+impl<Balance: Zero, Share: Zero> WithdrawStake<Balance, Share> {
```

Comment:
what about impl `Zero` trait for `WithdrawStake` instead?

Reply:
`Zero` trait requires another fn `fn zero() -> Self` which I'm not sure what to return and it still requires the trait bound `Balance: Zero, Share: Zero`, so doesn't seem necessary.